src/common/ClientHelper.ts

```diff
@@ -25,25 +33,22 @@ export async function constructSpokePoolClientsWithLookback(
   configStoreClient: AcrossConfigStoreClient,
   config: CommonConfig,
   baseSigner: Wallet,
-  initialLookBackOverride: { [chainId: number]: number } = {}
+  initialLookBackOverride = 0
 ): Promise<SpokePoolClientsByChain> {
   // Set up Spoke signers and connect them to spoke pool contract objects:
   const spokePoolSigners = getSpokePoolSigners(baseSigner, config);
   const spokePools = config.spokePoolChains.map((chainId) => {
     return { chainId, contract: getDeployedContract("SpokePool", chainId, spokePoolSigners[chainId]) };
   });
 
-  // For each spoke chain, look up its latest block and adjust by lookback configuration to determine
-  // fromBlock. If no lookback is set, fromBlock will be set to spoke pool's deployment block.
-  const fromBlocks: { [chainId: number]: number } = {};
-  const l2BlockNumbers = await Promise.all(
-    spokePools.map((obj: { contract: Contract }) => obj.contract.provider.getBlockNumber())
+  // If initial lookback override is non-zero, then look up the l2 block number at that time, otherwise
+  // use latest L2 block.
+  const _fromBlocks = await Promise.all(
+    initialLookBackOverride > 0
+      ? spokePools.map((obj) => utils.findBlockAtOrOlder(obj.contract.provider, initialLookBackOverride))
+      : spokePools.map((obj) => getDeployedBlockNumber("SpokePool", obj.chainId))
```

Comment:
The way you had it before your fix commit was good, I just think we needed some extra async/await keywords.

I think this should work to avoid two loops:
```suggestion
  const fromBlocks = Object.fromEntries(
    await Promise.all(
      initialLookBackOverride > 0
        ? spokePools.map(async (obj) => [
            obj.chainId,
            await utils.findBlockAtOrOlder(obj.contract.provider, initialLookBackOverride),
          ])
        : spokePools.map(async (obj) => [obj.chainId, await getDeployedBlockNumber("SpokePool", obj.chainId)])
    )
  );
```